index.js

```diff
@@ -194,16 +197,30 @@ class Channel {
     const s = this
     const typeLen = encodingLength(c.uint, type)
 
+    const self = this
+
     const m = {
       type,
       encoding,
       onmessage,
       recv (state, session) {
+        self.tracer.trace('recv', {
+          id: self.id,
+          protocol: self.protocol,
+          type
+        })
+
         return session._track(m.onmessage(encoding.decode(state), session))
       },
       send (m, session = s) {
         if (session.closed === true) return false
 
+        self.tracer.trace('send', {
+          id: self.id,
+          protocol: self.protocol,
+          type
+        })
```

Comment:
On line 197, `s = this` is already defined.

```suggestion
        s.tracer.trace('send', {
          id: s.id,
          protocol: s.protocol,
          type
        })
```

Reply:
Oops, fixed. Just pushed a commit instead of accepting the suggestion since I also had to update `recv`